sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_storage_utils.py

```diff
@@ -120,15 +120,32 @@ def get_uri_type(self) -> str:
 
 
 def get_storage_client(
-    credential: str,
-    storage_account: str,
+    *,
+    credential: Optional[str] = None,
+    storage_account: Optional[str] = None,
+    storage_client: Optional[Any] = None,  # TODO: support user passing in own storage client.
```

Comment:
We will need to allow the option of passing in a pre-configured/pre-authenticated Storage client.
If you let me know the public API areas where users are expected to configure their storage setup - I can take a shot at plumbing this through more extensively.